subt/crash_report.py

```diff
@@ -35,4 +35,60 @@ def update(self):
             assert False, channel  # unsupported channel
         return channel
 
+
+if __name__ == "__main__":
+    import argparse
+    import numpy as np
+    from osgar.lib.serialize import deserialize
+    from osgar.logger import LogReader, LogWriter, lookup_stream_id, lookup_stream_names
+
+    parser = argparse.ArgumentParser("Analyze crash_rgbd data")
+    parser.add_argument('logfile', help='path to logfile')
+    parser.add_argument('--verbose', '-v', help="verbose mode", action='store_true')
+    args = parser.parse_args()
+
+    crash_stream_id = lookup_stream_id(args.logfile, 'black_box.crash_rgbd')
+    pose3d_stream_id = lookup_stream_id(args.logfile, 'fromrospy.pose3d')
+
+    def nearest_pose(poses, pose3d):
+        xyz_arr = np.array([xyz for xyz, quat in poses])
+        xyz_diff = xyz_arr - pose3d[0]
+        dist = np.linalg.norm(xyz_diff, axis=1)
+        index = np.argmin(dist)
+        return index
+
+    stream_names = lookup_stream_names(args.logfile)
+
+    def get_camera_name_id(camera_pose):
+        x, y = camera_pose[0][:2]
+        if y == 0:
+            camera_direction = 'front' if x > 0 else 'rear'
+        elif y < 0:
+            camera_direction = 'right'
+        else:  # y > 0
+            camera_direction = 'left'
+        return stream_names.index(f'fromrospy.rgbd_{camera_direction}') + 1
+
+    poses = []
+    crash_time = None
+    with LogReader(args.logfile,
+               only_stream_id=[pose3d_stream_id, crash_stream_id]) as logreader, LogWriter(
+            filename='crash.log', start_time=logreader.start_time) as f:
```

Comment:
Can you make the crash log file name configurable with a commandline parameter, please?

Reply:
fixed:
```
(osgar) md@md-ThinkPad-P50:~/git/osgar$ ls -l /data/aws/local/subt-freyja-2021-08-06T21.57.04*
-rw-rw-r-- 1 md md 418093877 Aug  8 21:34 /data/aws/local/subt-freyja-2021-08-06T21.57.04-crash.log
-rw-r--r-- 1 md md 426364414 Aug  6 21:59 /data/aws/local/subt-freyja-2021-08-06T21.57.04.log
```
+ `--out` option